source/documentation/modules/color.md

```diff
@@ -0,0 +1,863 @@
+  * [`color.adjust()`](#adjust) for adjusting a color's properties by fixed
+  amounts.
+
+  {% codeExample 'color-change', false %}
+    @debug color.change(#6b717f, $red: 100); // #64717f
+    @debug color.change(#d2e1dd, $red: 100, $blue: 50); // #64e132
+    @debug color.change(#998099, $lightness: 30%, $alpha: 0.5); // rgba(85, 68, 85, 0.5)
+    ===
+    @debug color.change(#6b717f, $red: 100)  // #64717f
+    @debug color.change(#d2e1dd, $red: 100, $blue: 50)  // #64e132
+    @debug color.change(#998099, $lightness: 30%, $alpha: 0.5)  // rgba(85, 68, 85, 0.5)
+  {% endcodeExample %}
+{% endfunction %}
+
+{% function 'color.complement($color)', 'complement($color)', 'returns:color' %}
+  Returns the RGB [complement][] of `$color`.
+
+  This is identical to [`color.adjust($color, $hue: 180deg)`](#adjust).
+
+  [complement]: https://en.wikipedia.org/wiki/Complementary_colors
+
+  {% codeExample 'color-complement', false %}
+    // Hue 222deg becomes 42deg.
+    @debug color.complement(#6b717f); // #7f796b
+
+    // Hue 164deg becomes 344deg.
+    @debug color.complement(#d2e1dd); // #e1d2d6
+
+    // Hue 210deg becomes 30deg.
+    @debug color.complement(#036); // #663300
+    ===
+    // Hue 222deg becomes 42deg.
+    @debug color.complement(#6b717f)  // #7f796b
+
+    // Hue 164deg becomes 344deg.
+    @debug color.complement(#d2e1dd)  // #e1d2d6
+
+    // Hue 210deg becomes 30deg.
+    @debug color.complement(#036)  // #663300
+  {% endcodeExample %}
+{% endfunction %}
+
+{% function 'darken($color, $amount)', 'returns:color' %}
+  Makes `$color` darker.
+
+  The `$amount` must be a number between `0%` and `100%` (inclusive). Decreases
+  the HSL lightness of `$color` by that amount.
+
+  {% headsUp %}
+  {% # Un-indented because 4-space indentation causes extra <p> tags in Markdown %}
+  The `darken()` function decreases lightness by a fixed amount, which is often
+  not the desired effect. To make a color a certain percentage darker than it
+  was before, use [`color.scale()`](#scale) instead.
+
+  Because `darken()` is usually not the best way to make a color darker, it's
+  not included directly in the new module system. However, if you have to
+  preserve the existing behavior, `darken($color, $amount)` can be written
+  [`color.adjust($color, $lightness: -$amount)`](#adjust).
+
+  {% codeExample 'color-darken', false %}
+    // #036 has lightness 20%, so when darken() subtracts 30% it just returns black.
+    @debug darken(#036, 30%); // black
+
+    // scale() instead makes it 30% darker than it was originally.
+    @debug color.scale(#036, $lightness: -30%); // #002447
+    ===
+    // #036 has lightness 20%, so when darken() subtracts 30% it just returns black.
+    @debug darken(#036, 30%)  // black
+
+    // scale() instead makes it 30% darker than it was originally.
+    @debug color.scale(#036, $lightness: -30%)  // #002447
+  {% endcodeExample %}
+  {% endheadsUp %}
```

Comment:
@jerivas @sanajaved7 Thoughts on this unindented approach? I couldn't find a good workaround... Unless we leave this template as LiquidJS instead of Markdown?

The nested `{% codeExample %}` returns markup, which ruins the typical `stripIndent` approach used by the wrapping `{% headsUp %}` block, and the 4-space level of indentation then adds a wrapping `<p>` tag where we don't want one.

Reply:
I tried disabling markdown parsing inside `headsUp` for these cases and that seemed to work (no wrapping `<p>`), but it requires manually passing `false` every time to the shortcode. I toyed with the idea of disabling markdown parsing automatically if the current page is itself an `.md` file, and that seemed to work well too:

```typescript
type EleventyPage = {
  date?: string;
  inputPath?: string;
  fileSlug?: string;
  filePathStem?: string;
  outputFileExtension?: string;
  templateSyntax?: string;
  url?: string;
  outputPath?: string;
};

type ShortcodeData = {
  page: EleventyPage;
  ctx: Context;
  eleventy: {
    version: string;
    generator: string;
    env: Record<string, any>;
  };
};

const isMarkdown = (page: EleventyPage) =>
  Boolean(page?.inputPath?.endsWith('.md'));

/**
 * Returns HTML for a heads-up warning related to the main
 * documentation.
 */
export async function headsUp(
  this: ShortcodeData,
  contents: string,
  useMarkdown = true
) {
  const html = await liquidEngine.renderFile('heads_up', {
    contents: stripIndent(contents),
    useMarkdown: useMarkdown ?? isMarkdown(this.page),
  });
  return html.trim();
}
```

I like this approach because it relies on the context instead of trying to be clever with string replacements, but I'm happy with your latest approach too.

Reply:
@jerivas Thoughts on this approach? https://github.com/oddbird/sass-site/commit/e14edcdee6729e4a67b56b3ef30e22c556af6505